### PR TITLE
xsimd: 9.0.1 -> 11.1.0

### PR DESCRIPTION
**Pull request by doronbehar**
Diff: https://github.com/xtensor-stack/xsimd/compare/9.0.1...11.1.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
@ofborg build python3.pkgs.scipy

**Comment by doronbehar** [comment]
Let's see if CI will be green here.

**Comment by doronbehar** [comment]
It builds fine but now I noticed the tests aren't executed.

**Comment by doronbehar** [comment]
Oh wait, the tests run was very short, now it should be good.

**Comment by doronbehar** [comment]
@ofborg build python3.pkgs.scipy

**Comment by doronbehar** [comment]
But now xsimd tests fail :(. Now testing a patch.

@ofborg build xsimd

**Comment by doronbehar** [comment]
@ofborg build xsimd

File: pkgs/development/libraries/xsimd/default.nix [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+  version = "11.1.0";
   src = fetchFromGitHub {
     owner = "xtensor-stack";
     repo = "xsimd";
     rev = version;
-    sha256 = "sha256-onALN6agtrHWigtFlCeefD9CiRZI4Y690XTzy2UDnrk=";
+    sha256 = "sha256-l6IRzndjb95hIcFCCm8zmlNHWtKduqy2t/oml/9Xp+w=";
   };
+  patches = [
+    ./fix-exp10-darwin.patch
```
Comment (mweinelt) [inline pkgs/development/libraries/xsimd/default.nix:12]: Missing patch

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by mweinelt** [comment]
```
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_basic_math.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/doc/explicit_use_of_an_instruction_set_mean_tag_dispatch.cpp:1:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_batch.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:104: test/CMakeFiles/test_xsimd.dir/test_arch.cpp.o] Error 1
xsimd-x86_64-darwin> make[2]: *** Waiting for unfinished jobs....
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_batch_bool.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_batch_cast.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:90: test/CMakeFiles/test_xsimd.dir/test_api.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:118: test/CMakeFiles/test_xsimd.dir/test_basic_math.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:160: test/CMakeFiles/test_xsimd.dir/test_batch_cast.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/doc/CMakeFiles/test_doc_any_arch.dir/build.make:118: test/doc/CMakeFiles/test_doc_any_arch.dir/explicit_use_of_an_instruction_set_mean_tag_dispatch.cpp.o] Error 1
xsimd-x86_64-darwin> make[1]: *** [CMakeFiles/Makefile2:175: test/doc/CMakeFiles/test_doc_any_arch.dir/all] Error 2
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:146: test/CMakeFiles/test_xsimd.dir/test_batch_bool.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:132: test/CMakeFiles/test_xsimd.dir/test_batch.cpp.o] Error 1
xsimd-x86_64-darwin> make[1]: *** [CMakeFiles/Makefile2:120: test/CMakeFiles/test_xsimd.dir/all] Error 2
xsimd-x86_64-darwin> make: *** [Makefile:146: all] Error 2

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by doronbehar** [comment]
I don't understand why `Accelerate` won't propagate all of these dependencies :(.

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by doronbehar** [comment]
@ofborg build xsimd

**Comment by doronbehar** [comment]
I'm pretty sure this should be good, but Darwin CI seems stuck now from some reason, so I'll revisit this in ~12 hours from now (tomorrow in my time zone).

**Comment by risicle** [comment]
On macos 10.15 now:

```
cd /tmp/nix-build-xsimd-11.1.0.drv-0/source/build/test && ./test_xsimd
[doctest] doctest version is "2.4.11"
[doctest] run with "--help" for options
===============================================================================
/tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_xsimd_api.cpp:695:
TEST CASE:  [xsimd api | float types functions]<float>
  exp10

/tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_xsimd_api.cpp:466: ERROR: CHECK_EQ( extract(xsimd::exp10(T(val))), std::pow(value_type(10), val) ) is NOT correct!
  values: CHECK_EQ( 100, 100 )

===============================================================================
/tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_xsimd_api.cpp:695:
TEST CASE:  [xsimd api | float types functions]<double>
  exp10

/tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_xsimd_api.cpp:466: ERROR: CHECK_EQ( extract(xsimd::exp10(T(val))), std::pow(value_type(10), val) ) is NOT correct!
  values: CHECK_EQ( 100, 100 )

===============================================================================
[doctest] test cases:  301 |  299 passed | 2 failed | 0 skipped
[doctest] assertions: 7917 | 7915 passed | 2 failed |
[doctest] Status: FAILURE!
```

**Comment by risicle** [comment]
But built `python310Packages.scipy` successfully, nixos x86_64 & aarch64-linux

**Comment by doronbehar** [comment]
Thanks for the report @risicle . Indeed there's another upstream issue with the tests. I now disabled only the tests that fail on darwin, not by disabling their entire files, but with a patch. I tested the patch on Linux. So I hope it's OK now.

@ofborg build xsimd

**Comment by doronbehar** [comment]
x86_64-darwin build has succeded! This should be good to go, but I'll give someone else the final word.

**Comment by vcunat** [comment]
> If you upgrade `xsimd` to 11.0 the problem should disappear.

@rgommers: unfortunately this PR did not make the scipy error go away for me on x86 (lines with `exp10`).

We could try with CI as well
@ofBorg build python3.pkgs.scipy

**Comment by rgommers** [comment]
Hmm that's puzzling. If the build uses `pythran` 0.13.1 and `xsimd` 11, that's the same as we use upstream and in other places (conda-forge, other Linux distros). Although note that `xsimd` is vendored inside `pythran` and I don't think it's explicitly unvendored here, so I was already puzzled at how xsimd 9.0.1 could be picked up.

**Comment by vcunat** [comment]
Yes, that vendoring seems to be the cause here, at a glance.

**Review by yu-re-ka — APPROVED** [review]

**Comment by vcunat** [comment]
OK, scipy built for me with this on x86_64-darwin, after unvendoring in 4fa2f56bb86a

**Comment by vcunat** [comment]
Oh, I missed that aarch64-darwin build of xsimd was still broken :-(
https://hydra.nixos.org/build/229998544/nixlog/1/tail

**Comment by doronbehar** [comment]
Thanks for @yu-re-ka for the further investigation and testings, and for @vcunat for the unvendoring. A follow-up PR that attempts to disable (or fix) all failing tests is in: https://github.com/NixOS/nixpkgs/pull/246546

**Comment by vcunat** [comment]
IIRC aarch64-darwin is too overloaded on CI, so we can't get confirmation for that failing one.

**Comment by doronbehar** [comment]
> On macos 10.15 now:

@risicle do you have an m1 available? or is that a x86_64 darwin?

**Pull request merged**